Restoring the points when app is reloaded

CheshtaK opened the pull request:
### Description
The points are restored even after the user removes the app from the background. However this functionality has been implemented for the specific case when the user exits the app after the Memory Match Game. 
Please review the PR and if everything looks fine I'll proceed to implement the feature for other cases too.

Fixes #1317 

### Type of Change:

- Code


**Code/Quality Assurance Only**
- Bug fix (non-breaking change which fixes an issue)


### How Has This Been Tested?
I have tested the app on my Android device. Relevant GIF is provided as below.
![ezgif com-optimize 5](https://user-images.githubusercontent.com/32400008/50655034-d430ab00-0fb4-11e9-8cba-fc21e17f7400.gif)

### Checklist:

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas

**Code/Quality Assurance Only**
- [x] My changes generate no new warnings 
- [ ] My PR currently breaks something (fix or feature that would cause existing functionality to not work as expected)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been published in downstream modules

CheshtaK committed:
issue resolved

Rimjhim28 commented:
@CheshtaK The app is crashing when I am trying to reproduce the change.
Steps followed:
1. Play Home and School scenario from level 1
2. Exit the app
3. Remove the app from background
4. Open the app again and click on Load Game button
5. Click on Store
The app crashes!

![error](https://user-images.githubusercontent.com/26908195/51071865-08276280-167e-11e9-81e9-056361e26206.png)

Rimjhim28 reviewed (changes requested):
With the current code changes, the application is crashing.

CheshtaK commented:
@Rimjhim28 Hey! Sorry for the late reply! As I mentioned above I implemented the restore points feature solely for the case when the user exits the app after the Memory Match Game. However I've now implemented the feature for the other case too.

![ezgif com-optimize 4](https://user-images.githubusercontent.com/32400008/51403382-4fc65680-1b76-11e9-9245-2b36edfe4015.gif)

Rimjhim28 reviewed (changes requested):
@CheshtaK I am not sure about the changes that you have made. I am not able to reproduce the changes. The changes are not producing the desired effects.
Try doing this, complete home scene and score some points, migrate out of the app and remove it from the background, launch the app again, click on load game button and play the school scene. It is pretty evident that the points scored in home screen are lost.
Please describe in detail on how you are picturing this issue and how are you intending to solve it.

Rimjhim28 reviewed (commented):
Since the PR is very old and there has not been any interaction for long, I am closing the PR.
